sdk/typescript/src/cryptography/publickey.ts

```diff
@@ -24,41 +26,53 @@ export function bytesEqual(a: Uint8Array, b: Uint8Array) {
 /**
  * A public key
  */
-export interface PublicKey {
+export abstract class PublicKey {
 	/**
 	 * Checks if two public keys are equal
 	 */
-	equals(publicKey: PublicKey): boolean;
+	equals(publicKey: PublicKey) {
+		return bytesEqual(this.toBytes(), publicKey.toBytes());
+	}
 
 	/**
 	 * Return the base-64 representation of the public key
 	 */
-	toBase64(): string;
+	toBase64() {
+		return toB64(this.toBytes());
+	}
 
 	/**
-	 * Return the byte array representation of the public key
+	 * Return the base-64 representation of the public key
 	 */
-	toBytes(): Uint8Array;
+	toString() {
```

Comment:
I'm in favor of removing one of these implementations (either this or toBase64) given they do the same thing (mark as deprecated, remove later)

Reply:
Lets go with the base64 version, I don't think it's good for people to depend on custom toString() implementations, especially if they are expecting a specific format